app/src/components/ReleaseModal.tsx

```diff
@@ -0,0 +1,133 @@
+import { Button, Center, Group, Modal, Stack, TextInput, Tooltip } from '@mantine/core';
+import { DateTime } from 'luxon';
+import { useRouter } from 'next/router';
+import { useState } from 'react';
+import { AlertCircle, CircleCheck, InfoCircle } from 'tabler-icons-react';
+import { notifications } from '@mantine/notifications';
+
+export interface ReleaseModalProps {
+  open: boolean;
+  onClose: () => void;
+  id: string;
+  resourceType: ArtifactResourceType;
+}
+
+export default function ReleaseModal({ open = true, onClose, id, resourceType }: ReleaseModalProps) {
+  const router = useRouter();
+  const [version, setVersion] = useState('');
+
+  const { data: resource } = trpc.draft.getDraftById.useQuery({
+    id: id,
+    resourceType: resourceType
+  });
+  const ctx = trpc.useContext();
+  const deleteMutation = trpc.draft.deleteDraft.useMutation({
+    onSuccess: () => {
+      notifications.show({
+        title: `Draft ${resource?.resourceType} released!`,
+        message: `Draft ${resource?.resourceType}/${resource?.id} successfully released to the Measure Repository!`,
+        icon: <CircleCheck />,
+        color: 'green'
+      });
+      ctx.draft.getDraftCounts.invalidate();
+      ctx.draft.getDrafts.invalidate();
+    },
+    onError: e => {
+      console.error(e);
+      notifications.show({
+        title: `Release Failed!`,
+        message: `Attempt to release ${resourceType} failed with message: ${e.message}`,
+        icon: <AlertCircle />,
+        color: 'red'
+      });
+    }
+  });
+
+  async function confirm() {
+    // requirements:
+    // https://build.fhir.org/ig/HL7/cqf-measures/measure-repository-service.html#release
+    // TODO: release recursively all children (ignore for now).
+    if (resource) {
+      resource.version = version;
+      resource.status = 'active';
+      resource.date = DateTime.now().toISO() || '';
+    }
+
+    const res = await fetch(`${process.env.NEXT_PUBLIC_MRS_SERVER}/${resourceType}`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json+fhir'
+      },
+      body: JSON.stringify(resource)
+    });
+
+    const location = res.headers.get('Location')?.substring(5); // remove 4_0_1 (version)
+    if (res.status !== 201) {
+      console.error(res.statusText);
+      notifications.show({
+        title: `Release Failed!`,
+        message: `Server unable to process request`,
+        icon: <AlertCircle />,
+        color: 'red'
+      });
+    } else if (!location) {
+      console.error('No resource location for released artifact');
+      notifications.show({
+        title: `Release Failed!`,
+        message: `No resource location exists for draft artifact`,
+        icon: <AlertCircle />,
+        color: 'red'
+      });
+    } else {
+      // delete draft
+      deleteMutation.mutate({
+        resourceType: resourceType,
+        id: id
+      });
+
+      // direct user to published artifact detail page
+      router.push(location);
```

Comment:
I agree this needs to be looked at. I think this should be done in a separate task since it looks like this is happening when the user just navigates to an artifact on the publishable measure repository, so it is a bit out of scope for this task.